### PR TITLE
Remove `RewriteTensorPointer` from the optimization pipeline

### PR DESCRIPTION
**Pull request by etiotto**

### CODE REVIEW

**Comment by etiotto** [comment]
Depends on PR #2598

**Comment by etiotto** [comment]
Performance results (see lablel "etiotto3":

http://benchmarks.glados.intel.com/d/1pXX4hUSz/microbenchmarks?orgId=1&var-tag=ci%7Cetiotto3&var-bench=attn&var-bench=gemm&var-bench=gemm-preop-exp&var-bench=gemm-postop-gelu&var-bench=gemm-postop-addmatrix&var-bench=gemm-streamk&var-bench=gemm-splitk&var-bench=gemm-bt&var-bench=gemm-at&var-device=All&var-compiler=triton&var-backend=All&var-baseline_backend=triton-ci-XPU%201550&var-target_backend=triton-ci-XPU%201550&from=now-5d&to=now

There was a degradation in "gemm-at" which has been fixed in this PR by enhancing the lowering code (to LLVM dialect" for loads with blocked pointers so that we can now generate vector loads of the appropriate vector length.

File: third_party/intel/lib/Analysis/AxisInfo.cpp [hunk 1]
```diff
@@ -558,6 +558,11 @@ class LoadOpAxisInfoVisitor final : public AxisInfoVisitorImpl<triton::LoadOp> {
     // If pointers and mask both have constancy properties, those properties
     // will also extend to output.
     AxisInfo ptrInfo = operands[0]->getValue();
+
+    llvm::errs() << "ptrInfo: ";
```
Comment (etiotto) [inline third_party/intel/lib/Analysis/AxisInfo.cpp:562]: Remove

**Comment by etiotto** [comment]
> Depends on PR #2598

I have merged that PR in now.

File: python/test/unit/language/test_block_pointer.py [hunk 2]
```diff
@@ -7,51 +7,53 @@
 
 
 @triton.jit
-def block_copy_kernel(a_ptr, b_ptr, N, BLOCK_SIZE: tl.constexpr, padding_option: tl.constexpr):
+def block_copy_kernel(a_ptr, b_ptr, N, BLOCK_SIZE: tl.constexpr):
```
Comment (etiotto) [inline python/test/unit/language/test_block_pointer.py:10]: Undo these changes. Checked in by mistake.

Comment (etiotto) [reply python/test/unit/language/test_block_pointer.py]: removed

**Review by whitneywhtsang — COMMENTED** [review]
Can you please explain why all changes not in compiler.py are needed?

**Comment by etiotto** [comment]
> Can you please explain why all changes not in compiler.py are needed?

You want  some comments in the PR to help understand what each change does ?

File: third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp [hunk 3]
```diff
@@ -818,49 +823,38 @@ struct LoadOpConversion
   LogicalResult
   matchAndRewrite(triton::LoadOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    auto loc = op->getLoc();
-    auto typeConverter = getTypeConverter();
-    auto *ctx = rewriter.getContext();
+    if (isTensorPointerType(op.getPtr().getType()) &&
```
[the remark this answers is not in the document]
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp:826]: I pulled some changes out in PR #2623

**Review by chengjunlu — APPROVED** [review]
